src/internal/DefaultInputState.ts

```diff
@@ -32,7 +32,17 @@ export class DefaultInputState implements InputState {
             ++offset;
         }
         const is = new DefaultInputState(this.ism, offset);
-        return { skipped: is.seenSince(this), state: is };
+        const skipped = is.seenSince(this);
+        let newListeners;
+        if (this.listeners) {
+            newListeners = cloneListeners(this.listeners);
+            const listeners: InputStateListener[] = Object.getOwnPropertyNames(newListeners).map(key => newListeners[key]);
```

Comment:
pretty sure we want Object.keys() here
in case someone has added a non-enumerable field, we don't want to get that